akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -135,6 +135,7 @@ abstract class SerializerWithStringManifest extends Serializer {
    * and message is dropped. Other exceptions will tear down the TCP connection
    * because it can be an indication of corrupt bytes from the underlying transport.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
Indeed it's compatible for implementations of the class, as "throwing less exceptions" is OK. 

AFAICS it's not compatible for code calling this method, as that code now has to take into account the fact that this method may throw, and `NotSerializableException` is a checked exception, not a `RuntimeException`.

I suppose we don't expect these methods to be called from client code? (hmm https://github.com/scullxbones/akka-persistence-mongo/blob/68b142589e23d5c9795220a8dc6acf889571c1b1/common/src/main/scala/akka/contrib/persistence/mongodb/MongoDataModel.scala#L70 does use `toBinary` directly but AFAICS not `fromBinary`, funny)